frontend/src/actions/story.js

```diff
@@ -1,21 +1,31 @@
-import {SAVE_STORY_START, SAVED_STORY, SUCCESS} from './types'
+import {SAVE_STORY_START, SAVED_STORY, SUCCESS, SIGN_IN} from './types'
 import api from 'utils';
 import history from '../history';
 
-export const submitStory = (dto) => async (dispatch) => {
+export const submitStory = (dto) => async (dispatch, getState) => {
+    const { token } = getState().auth;
     dispatch({type: SAVE_STORY_START});
     const {story, nextPage} = dto;
     const response = await api(`stories/`, {
         method: 'POST',
         body: story,
-      });
+    });
       
     dispatch({
-    type: SAVED_STORY,
-    payload: {
-        status: response.error || { type: SUCCESS },
-        story: (!response.error && response) || null,
-    }
+        type: SAVED_STORY,
+        payload: {
+            status: response.error || { type: SUCCESS },
+            story: (!response.error && response) || null,
+        }
+    });
+
+    // if there was no token identifying a user, we need to store the story auth for future data validation
+    if (!token) dispatch({
+        type: SIGN_IN,
+        payload: {
+            status: {type: SUCCESS},
+            token,
+        }
```

Comment:
@anaPerezGhiglia here's where the frontend stores the story auth token! Check it out:

- On line 6 we fetch whatever's in the store for auth tokens right now
- Then on line 22, if we didn't have a token (meaning, the user wasn't authed), we dispatch a sign in action with the token!
- You can also check the changes in the `utils` file which enforce the use of this token in any subsequent api calls, so, if there's a token available, it WILL be used - even if we're calling a public endpoint.